src/test/shell/bazel/remote/remote_execution_test.sh

```diff
@@ -1056,6 +1056,33 @@ EOF
   expect_not_log "remote cache hit"
 }
 
+function test_remote_cache_upload() {
+  mkdir -p a
+  echo 'bar' > a/bar
+  cat > a/BUILD <<'EOF'
+genrule(
+  name = "foo",
+  srcs = [":bar"],
+  outs = ["foo.txt"],
+  cmd = """
+    echo $(location :bar) > "$@"
+  """,
+)
+EOF
+
+  CACHEDIR=$(mktemp -d)
+
+  bazel build \
+    --disk_cache=$CACHEDIR \
+    --remote_cache=grpc://localhost:${worker_port} \
+    //a:foo >& $TEST_log || "Failed to build //a:foo"
+
+  remote_ac_files="$(count_remote_ac_files)"
+  [[ "$remote_ac_files" == 1 ]] || fail "Expected 1 remote action cache entries, not $remote_ac_files"
```

Comment:
Before this PR, this would be 0 indicating cache misses.

Worth to note that this behavior could only be observed when both `--disk_cache` AND `--remote_cache` are both enabled, thus triggering `DiskAndRemoteCacheClient.java` logic.

Personally I would consider enabling both `--disk_cache` and `--remote_cache` somewhat of a miss-use since enabling `--remote_cache` alone would keep a copy of cas and ac on local disk within execroot 🤔. But perhaps we can discuss/document the tradeoffs between these 2 in a separate change

Reply:
```
bazel test src/test/shell/bazel/remote:remote_execution_test --test_arg=test_remote_cache_upload
```

Handy command for those who are not familiar with Bazel code base's shell test setup to try out the test and see the result with/without the one-liner fix.